### PR TITLE
Add more rules to markdownlint

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Add more rules to markdownlint and add a `Makefile` with `mdl` target.

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+markdownlint: mdl
+
+mdl:
+	mdl --rules \
+		MD002,MD003,MD004,MD005,MD006,MD007,MD008,MD009,MD010,MD011,MD012,MD014,MD018,MD019,MD020,MD021,MD022,MD023,MD024,MD026,MD027,MD028,MD029 \
```
Comment (sumo-drosiek) [inline Makefile:5]: What do we skip?

File: deploy/docs/Performance.md [hunk 2]
```diff
@@ -5,9 +5,9 @@ For larger or more volatile loads, we recommend [enabling Fluentd autoscaling](.
 ## Recommendations
 
 1. At least **8 Fluentd-logs** pods per **1 TB/day** of logs.
-2. At least **4 Fluentd-metrics** pods per **120k DPM** of metrics.
-3. The Prometheus pod will use on average **2GiB** memory per **120k DPM**; however in our experience this has gone up to **5GiB**, so we recommend allocating ample memory resources for the Prometheus pod if you wish to collect a high volume of metrics for a larger cluster.
-4. For clusters with 500 application pods or greater, we found the following configuration changes in the [`values.yaml`](./../helm/sumologic/values.yaml) file to lead to a more stable experience:
+1. At least **4 Fluentd-metrics** pods per **120k DPM** of metrics.
+1. The Prometheus pod will use on average **2GiB** memory per **120k DPM**; however in our experience this has gone up to **5GiB**, so we recommend allocating ample memory resources for the Prometheus pod if you wish to collect a high volume of metrics for a larger cluster.
+1. For clusters with 500 application pods or greater, we found the following configuration changes in the [`values.yaml`](./../helm/sumologic/values.yaml) file to lead to a more stable experience:
```
Comment (sumo-drosiek) [inline deploy/docs/Performance.md:10]: Pretty strange IMHO

Comment (pmalek-sumo) [reply deploy/docs/Performance.md:10]: This violated `MD029 - Ordered list item prefix`

```
deploy/docs/Performance.md:8: MD029 Ordered list item prefix
deploy/docs/Performance.md:9: MD029 Ordered list item prefix
deploy/docs/Performance.md:10: MD029 Ordered list item prefix
deploy/docs/Performance.md:23: MD029 Ordered list item prefix
deploy/docs/Performance.md:24: MD029 Ordered list item prefix
```

File: Makefile (hunk 1 above)
Comment (pmalek-sumo) [reply Makefile:5]: Just a couple. I can either work on this here or reiterate again in another PR.
The most problematic will be the line length `MD013 - Line length` as there are many occurrences of its violation.

File: deploy/docs/Performance.md (hunk 2 above)
Comment (sumo-drosiek) [reply deploy/docs/Performance.md:10]: 🤯

File: .markdownlint/style.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+rule 'MD028'
+rule 'MD029'
+rule 'MD030'
+rule 'MD031'
+rule 'MD032'
+rule 'MD035'
+rule 'MD037'
+rule 'MD038'
+rule 'MD039'
+rule 'MD046'
```
Comment (sumo-drosiek) [inline .markdownlint/style.rb:31]: Let's add names of the rules or link to them in a comment in the header or both :)

https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md

Comment (pmalek-sumo) [reply .markdownlint/style.rb:33]: Done, added the link to these docs

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+markdownlint: mdl
+
+mdl:
+	mdl --style .markdownlint/style.rb deploy/docs
```
Comment (sumo-drosiek) [inline Makefile:4]: same ask like for `yamllint`, please add it to vagrant's Makefile, so we can use `sumo-make`

**Review by sumo-drosiek — APPROVED** [review]
LGTM

**Review by sumo-drosiek — CHANGES_REQUESTED** [review]
Does the vagrant's `sumo-make yamllint` works fine?

**Comment by pmalek-sumo** [comment]
> Does the vagrant's `sumo-make yamllint` works fine?

Now it works with `${root_dir}`

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**